### ISSUE TITLE
Yolov8 Docker Image >v8.2.27 not saving Results anymore

### ISSUE THREAD

**Issue by Idefix0496** [ultralytics/ultralytics#16425]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train, Predict, Multi-GPU

### Bug

Hi,
so a few days ago I tried predicting and training with the Yolov8 Docker Image in Docker desktop again. For that purpose I downloaded the newest Image and created a Container as usual.
However as soon as I started predicting or training I noticed that I've no longer get the results Saved to the Container resp. my Physical Disk anymore. The **runs** directory is empty
It says **Results saved to /usr/src/ultralytics/runs/detect/predict** as usual but it is deffinetely not. 
Even going to the path **/usr/src/ultralytics/runs/detect/predict** Manually via PowerShell and checking the files, nothing is there.
Going back to an old Image v8.2.27 everything still works fine.
This behavior is the same across training and predicting

So did something change in the subsecuent revisons that I need to consider or is it just a Bug ?

### Environment

**Working Image:**
Ultralytics YOLOv8.2.27 🚀 Python-3.10.14 torch-2.2.2 CUDA:0 (NVIDIA GeForce GTX 1080 Ti, 11264MiB)
Setup complete ✅ (24 CPUs, 31.2 GB RAM, 87.5/250.9 GB disk)

OS                  Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Environment         Docker
Python              3.10.14
Install             git
RAM                 31.19 GB
CPU                 Intel Core(TM) i9-10920X 3.50GHz
CUDA                12.1

matplotlib          ✅ 3.9.0>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.2.2>=1.8.0
torchvision         ✅ 0.17.2>=0.9.0
tqdm                ✅ 4.65.0>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0

**Faulty Image:**
Ultralytics YOLOv8.2.98 🚀 Python-3.10.14 torch-2.3.1 CUDA:0 (NVIDIA GeForce GTX 1080 Ti, 11264MiB)
Setup complete ✅ (24 CPUs, 31.2 GB RAM, 87.5/250.9 GB disk)

OS                  Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Environment         Docker
Python              3.10.14
Install             git
RAM                 31.19 GB
CPU                 AMD EPYC 7763 64-Core Processor
CUDA                12.1

numpy               ✅ 1.23.5<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.2>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.3.1>=1.8.0
torchvision         ✅ 0.18.1>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.7>=2.0.0
torch               ✅ 2.3.1!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

docker image pull ultralytics/ultralytics:latest

docker run --ipc=host -it -v E:\Pytorch\Datasets:/usr/src/datasets -v E:\Pytorch\Results:/usr/src/ultralytics/runs  --gpus all 3aa413994eec

yolo detect predict save=True model=yolov8l.pt source=/usr/src/datasets/testing/Fotos/ imgsz=640 device=1

### Additional

It seems like something is wrong with the **yolo checks** Function.
Although it would be really nice to have an **AMD EPYC 7763 64-Core Processor** the reallity is more near a **Intel Core(TM) i9-10920X 3.50GHz** Processor.
So I don't know if this is related to my issue. 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Idefix0496, thank you for bringing this up and for your interest in Ultralytics 🚀! This is an automated response, but rest assured, an Ultralytics engineer will be with you shortly to assist further.

We suggest visiting the [Docs](https://docs.ultralytics.com) for more information, where you can find helpful usage examples for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/), which might already have answers to some of your questions.

If this issue is a 🐛 Bug Report, please ensure you’ve provided all the necessary details and consider sharing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to aid our debugging efforts.

Should you have any ❓ Questions regarding custom training, it would be helpful to include detailed information, such as dataset image examples and training logs. Also, please confirm you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To jog your experience with the Ultralytics community, hop onto our [Discord](https://ultralytics.com/discord) for real-time conversations 🎧, or engage in more in-depth discussions over at [Discourse](https://community.ultralytics.com). Don't forget our [Subreddit](https://reddit.com/r/ultralytics) for community interactions.

## Upgrade

Make sure you're using the latest `ultralytics` package to see if the issue persists:

```bash
pip install -U ultralytics
```

## Verified Environments

YOLOv8 operates nicely in these environments, all dependencies included ([CUDA](https://developer.nvidia.com/cuda), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/)):

- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**, **AWS**, and **Docker** setups – guides are available.

## Current Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the CI badge is green, our tests are passing smoothly. This ensures correct operation across all platforms including macOS, Windows, and Ubuntu every 24 hours and with every commit.

Stay tuned! 🎉 Your issue will receive a detailed follow-up soon.

**Comment by Y-T-G** [comment]
It works fine for me with the latest image. The `ultralytics` folder inside the docker image is at `/ultralytics`.

**Comment by Y-T-G** [comment]
The `WORK_DIR` was changed recently, so you're probably looking at the wrong folder.

https://github.com/ultralytics/ultralytics/commit/01977dac413fcc6a7a9d2e3f5bd1cffb791e7cfb

It's at `/ultralytics` now and not `/usr/src/ultralytics`

**Comment by ambitious-octopus** [comment]
Hey @Idefix0496, I was able to reproduce this, the problem is because persistent cache introduced in:  https://github.com/ultralytics/ultralytics/pull/16427
Working on this!

**Comment by Idefix0496** [comment]
Hey Thanks for the quick replies.

@Y-T-G 
Yeah changing the WORK_DIR solved the Problem for me.

**I changed the CLI-Inputs as following:**
docker run --ipc=host -it -v E:\Pytorch\Results:/ultralytics/runs -v E:\Pytorch\Datasets:/ultralytics/datasets  --gpus all a162b84c6924
yolo detect predict model=yolov8l.pt source=/ultralytics/datasets/testing/Fotos/ imgsz=640 device=1

Now I can Access the Results over at this path:
/ultralytics/runs/detect/predict## or direct in Windows E:\Pytorch\Results\detect\predict##

@ambitious-octopus 
That's odd because with the changes I made it is working now. 
Also tested with v8.2.100

**Comment by ambitious-octopus** [comment]
@Idefix0496 yes, this is more related to the` yolo check` problem.
